### PR TITLE
Bring back Storybooks

### PR DESCRIPTION
**Pull request by orta**
screen shot 2017-05-01 at 17 42 19

Migrates a lot of the setup inside reaction-force into Emission, as Storybooks were removed in #453 

Note: this include the fusebox manager, should we keep this? I'm close to having the webpack setup down. So once that's working I could look at fusebox instead.

*TODO*:

- [x] Fix Dangerfile
- [x] Port rest of books to TS
- [x] Clean up the folder structure
- [ ] Figure out if webpack is fast enough
- [ ] Fusebox? 
- [ ] Storysnaps?

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Running your Dangerfile has Failed</td>
    </tr>
  </tbody>
</table>




## Danger has errored

Error: TypeError

```sh
TypeError: Cannot read property 'length' of null
    at reactComponentForPath (/home/travis/build/artsy/dangerfile.ts:87:12)
    at Array.map (native)
    at Object.<anonymous> (/home/travis/build/artsy/dangerfile.ts:94:52)
    at Runtime._execModule (/home/travis/build/artsy/emission/node_modules/jest-runtime/build/index.js:447:13)
    at Runtime.requireModule (/home/travis/build/artsy/emission/node_modules/jest-runtime/build/index.js:295:14)
    at /home/travis/build/artsy/emission/node_modules/danger/source/runner/DangerfileRunner.ts:92:13
    at ensureCleanDangerfile (/home/travis/build/artsy/emission/node_modules/danger/source/runner/DangerfileRunner.ts:129:3)
    at Object.<anonymous> (/home/travis/build/artsy/emission/node_modules/danger/source/runner/DangerfileRunner.ts:91:3)
    at step (/home/travis/build/artsy/emission/node_modules/danger/distribution/runner/DangerfileRunner.js:32:23)
    at Object.next (/home/travis/build/artsy/emission/node_modules/danger/distribution/runner/DangerfileRunner.js:13:53)
```


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by orta** [comment]
OK, storybooks are back in and I've added a few more

**Comment by orta** [comment]
Gonna bail on fusebox etc for now, and have a separate PR for Danger checking that people are including stories for new components.

File: dangerfile.ts [hunk 1]
```diff
@@ -83,21 +89,19 @@ if (testFilesThatDontExist.length > 0) {
...
 // If any are left, leave a warning.
 let componentsForFiles = compact(touchedComponents.map(reactComponentForPath))
-
-// This may need updating once there are multiple folders for components
-const storyFiles = fs.readdirSync("src/stories")
+const storyFiles = allFiles.filter(f => f.includes("__stories__/") && f.includes(".story."))
 
 storyFiles.forEach(story => {
-  const content = fs.readFileSync("src/stories/" + story).toString()
+  const content = fs.readFileSync(story, "utf8")
```
Comment (alloy) [inline dangerfile.ts]: Oooh, you can leave out the `{ encoding: ... }` part?!

**Comment by alloy** [comment]
This may already be clear to you, but we’re no longer using fusebox in reaction-force. So this look great to me, good stuff :clap:

File: dangerfile.ts (hunk 1 above)
Comment (orta) [inline dangerfile.ts]: Yep

screen shot 2017-05-02 at 22 16 20

**Comment by orta** [comment]
I didn't look :D

File: package.json [hunk 2]
```diff
@@ -68,6 +68,7 @@
     "lib"
   ],
   "dependencies": {
+    "awesome-typescript-loader": "^3.1.3",
```
Comment (orta) [inline package.json]: This is an issue

Comment (alloy) [inline package.json]: Ah, again :)

Comment (orta) [inline package.json]: For others not in our slack, I just learned that React Native uses the haste-map when bundling to the App - which means that TypeScript and awesome-typescript-loader are not shipped to Eigen

Comment (orta) [inline package.json]: [context](https://artsy.slack.com/archives/C02BAQ5K7/p1493759959900587)

**Pull request merged**